### DISCUSSION TITLE
Does tap action service call support templates?

### DISCUSSION THREAD

**Discussion by totalitarian** [piitaya/lovelace-mushroom#514] [Q&A]
Hoping to implement something like so but get an error saying required key data[oscillating] missing

```
        tap_action:
          action: call-service
          service: fan.oscillate
          data: >
            {% if state_attr('fan.living_room_fan', 'oscillating') %}
              {'oscillating': false }
            {% else %}
              {'oscillating': true }
            {% endif %}
```

**Comment by piitaya** [comment]
It's not supported and will not be supported because action is a complete object.
But you can easily doing this by creating a script ! 😊

**Reply by krazos** [reply]
Is this a technical limitation, and is it specific to templates within the `call-service` action?  I have a slightly different use case where a script isn't really helpful:

I have an entity card for my TV.  If the TV is off, I want the `tap_action` to turn on the TV, but if the TV is already on, I want the `tap_action` to navigate to another view with a custom remote control card.  I accomplish this currently by using two conditional cards for the TV that are identical but for the `tap_action`, but this is a bit clunky...

**Reply by piitaya** [reply]
Yes it's a technical limitation. Template is string and is rendered server side. Tap action is JSON object are created front side. It will be too complex to add this support and I do not prefer to add complexity to Mushroom. 
Using conditional card is preferable.
